### PR TITLE
Fixed cookies parsing bug in auto captured cookies

### PR DESCRIPTION
**Pull request by ninadbstack**
Context:
- Cookie parsing logic failed when cookies were encoded as base64 and had `=` at the end. 
- When servers used strict base64 cookie check this ended up with error response causing `ERR_HTTP2_PROTOCOL_ERROR` in discovery chromium. 

This PR fixes the logic and refactors the function. 
Also few unrelated logging improvements.

Secure cookie ->
- Sometime we are getting `Protocol error (Network.setCookies): Invalid cookie fields` because the percy captured cookie did not capture the secure flag
- Adding support for secure flag that can be set by setting this env variable `PERCY_SECURE_COOKIES` to true.

### CODE REVIEW

**Review by prklm10 — APPROVED** [review]

**Review by pankaj443 — APPROVED** [review]

File: packages/core/src/utils.js [hunk 1]
```diff
@@ -461,6 +461,27 @@ export function snapshotLogName(name, meta) {
...
+  ) return null;
+
+  return cookiesStr.split('; ').map(c => {
+    const eqIdx = c.indexOf('=');
+    const cookieObj = {
+      name: c.substring(0, eqIdx),
+      value: c.substring(eqIdx + 1)
+    };
+    if (process.env.PERCY_SECURE_COOKIES === 'true') {
+      cookieObj.secure = true;
```
Comment (ninadbstack) [inline packages/core/src/utils.js:479]: Why is this required? also we cant capture secure cookies, how can we set those here?

Comment (prklm10) [reply packages/core/src/utils.js:479]: It is because for one of the customer setting  captured cookie was throwing error. I debugged it and found this. setting the cookie to secure solves the issue.
I have added support for this through env variable so it does not affect other cookie.

Comment (ninadbstack) [reply packages/core/src/utils.js:479]: Wouldnt it affect all the cookies ? All will become secure. And secure cookies. Ideally we should set these to same as in test browser. 

Please add a comment on this that new cookie capture process should update this code.

Comment (prklm10) [reply packages/core/src/utils.js]: reverted my changes. And i will be sending it in separate pr.
